troposphere/awslambda.py

```diff
@@ -38,6 +38,10 @@ def validate(self):
             raise ValueError(
                 "You can't specify both 'S3ObjectVersion' and 'ZipFile'"
             )
+        if zip_file and len(zip_file) > 4096:
```

Comment:
Did you think about how this would work if you write this with a Join? I'll give an example below. I think this check is great if `zip_file` is a string, but we shouldn't assume it is.

```
ZipFile=Join('', [
    "var x = '", Ref(InputXParam), "';\n",
   "var y = '", Ref(InputYParam), "';\n",
   rest_of_code_var,
])
```